go-controller/pkg/node/controllers/egressip/egressip_test.go

```diff
@@ -229,6 +229,21 @@ func setupFakeTestNode(nodeInitialState nodeConfig) (ns.NetNS, cleanupFn, error)
 	return testNS, cleanupFn, nil
 }
 
+func createVRFAndEnslaveLink(testNS ns.NetNS, linkName, vrfName string, vrfTable int) error {
+	return testNS.Do(func(netNS ns.NetNS) error {
+		exec := kexec.New()
+		out, err := exec.Command("ip", "link", "add", vrfName, "type", "vrf", "table", fmt.Sprintf("%d", vrfTable)).CombinedOutput()
+		if len(out) > 0 || err != nil {
+			return fmt.Errorf("failed to create VRF: err: %v, out: %v", err, string(out))
+		}
+		out, err = exec.Command("ip", "link", "set", "dev", linkName, "master", vrfName).CombinedOutput()
+		if len(out) > 0 || err != nil {
+			return fmt.Errorf("failed to add link %s to VRF %s: err: %v, out: %v", linkName, vrfName, err, string(out))
+		}
```

Comment:
Just curious, would you want to use `netlink.LinkAdd` and `netlink.LinkSetMaster` here instead? :thinking: